include/seqan3/utility/views/zip.hpp

```diff
@@ -6,26 +6,481 @@
+template <bool Const>
+#if defined(__GNUC__) && (__GNUC__ == 10) // cpp17 iterators
+class zip_view<Views...>::iterator : public zip::iterator_category_t<Const, Views...>
+#else // cpp20 iterators
+class zip_view<Views...>::iterator : public zip::iterator_category_t<zip::all_forward<Const, Views...>>
+#endif
+{
+private:
+    constexpr explicit iterator(
+        zip::tuple_or_pair<std::ranges::iterator_t<view_helper::maybe_const<Const, Views>>...> current)
+    : current_(std::move(current))
+    {}
+
+    friend class zip_view<Views...>;
+
+public:
+    // Exposition-only. Is public for access via friend operator== of the sentinel.
+    zip::tuple_or_pair<std::ranges::iterator_t<view_helper::maybe_const<Const, Views>>...> current_;
+
+    using iterator_concept =
+        std::conditional_t<zip::all_random_access<Const, Views...>, std::random_access_iterator_tag,
+            std::conditional_t<zip::all_bidirectional<Const, Views...>, std::bidirectional_iterator_tag,
+                std::conditional_t<zip::all_forward<Const, Views...>, std::forward_iterator_tag,
+                    std::input_iterator_tag>>>;
+    using value_type = zip::tuple_or_pair<std::ranges::range_value_t<view_helper::maybe_const<Const, Views>>...>;
+    using difference_type = std::common_type_t<
+                                std::ranges::range_difference_t<view_helper::maybe_const<Const, Views>>...>;
+
+    iterator() = default;
+    constexpr iterator(iterator<!Const> i)
+        requires Const && (std::convertible_to<std::ranges::iterator_t<Views>,
+                                               std::ranges::iterator_t<view_helper::maybe_const<Const, Views>>> && ...)
+    : current_(std::move(i.current))
+    {}
+
+    constexpr auto operator*() const
+    {
+        return zip::tuple_transform([] (auto & i) -> decltype(auto) { return *i; }, current_);
+    }
+    constexpr iterator& operator++()
+    {
+        zip::tuple_for_each([] (auto & i) { ++i; }, current_);
+        return *this;
+    }
+    constexpr void operator++(int)
+    {
+        ++*this;
+    }
+    constexpr iterator operator++(int)
+        requires zip::all_forward<Const, Views...>
+    {
+        auto tmp = *this;
+        ++*this;
+        return tmp;
+    }
+
+    constexpr iterator& operator--()
+        requires zip::all_bidirectional<Const, Views...>
+    {
+        zip::tuple_for_each([] (auto & i) { --i; }, current_);
+        return *this;
+    }
+    constexpr iterator operator--(int)
+        requires zip::all_bidirectional<Const, Views...>
+    {
+        auto tmp = *this;
+        --*this;
+        return tmp;
+    }
+
+    constexpr iterator& operator+=(difference_type x)
+        requires zip::all_random_access<Const, Views...>
+    {
+        zip::tuple_for_each([&] <typename I> (I & i) { i += std::iter_difference_t<I>(x); }, current_);
+        return *this;
+    }
+    constexpr iterator& operator-=(difference_type x)
```

Comment:
```suggestion
    }

    constexpr iterator& operator-=(difference_type x)
```

Reply:
Do you want the spaces everywhere or just here?

Reply:
everywhere. We need clang format :D